--- hydra-4.1-src/configure.orig        2004-06-20 15:19:04.190641728 +0000
+++ hydra-4.1-src/configure     2004-06-20 15:18:41.026163264 +0000
@@ -244,9 +244,9 @@
 if [ -n "$CRYPTO_PATH" ]; then
     XLIBS="$XLIBS -lcrypto"
 fi
-if [ -n "$NSL_PATH" ]; then
-    XLIBS="$XLIBS -lnsl"
-fi
+#if [ -n "$NSL_PATH" ]; then
+#    XLIBS="$XLIBS -lnsl"
+#fi
 if [ -n "$SOCKET_PATH" ]; then
     XLIBS="$XLIBS -lsocket"
 fi

