--- thc-hydra-9.4/configure~	2022-09-08 08:32:44.000000000 +0000
+++ thc-hydra-9.4/configure	2022-11-04 18:41:21.454365669 +0000
@@ -153,9 +153,7 @@
   SDK_PATH=`xcrun --show-sdk-path`
 fi
 LIBDIRS=`cat /etc/ld.so.conf /etc/ld.so.conf.d/* 2> /dev/null | grep -v '^#' | sort | uniq`
-if [ "$SIXFOUR" = "64" ]; then
-  LIBDIRS="$LIBDIRS /lib64 /usr/lib64 /usr/local/lib64 /opt/local/lib64"
-fi
+LIBDIRS="$LIBDIRS /libx32 /usr/libx32 /usr/local/libx32 /opt/local/libx32"
 if [ -d "/Library/Developer/CommandLineTools/usr/lib" ]; then
   LIBDIRS="$LIBDIRS /Library/Developer/CommandLineTools/usr/lib /Library/Developer/CommandLineTools/lib"
 fi
